### ISSUE TITLE
ESP-IDF-4 not working on DC or mobile battery ( without usb serial) at debug build.

### ISSUE THREAD

**Issue by MKGaru** [Moddable-OpenSource/moddable#544]
branch: esp-idf-4
version: e01b04ce2
platform: esp32
command: `mcconfig -d -m -p esp32`

ESP-IDF-4 not working on DC or mobile battery ( without usb serial) at debug build.

but can work without debug build option: `mcconfig -m -p esp32`
Is this behavior correct?

manifest.json
```json
{
	"include": [
		"$(MODDABLE)/examples/manifest_base.json",
		"$(MODULES)/pins/digital/manifest.json"
	],
	"modules": {
		"*": "./main"
	}
}
```

main.js
```js
import Digital from "pins/digital"

Digital.write(2, true)
```

public branch with debug build without usb serial

https://user-images.githubusercontent.com/6237028/104805882-9ce96b80-5816-11eb-9219-3f9f0b6f37ee.mp4

esp-idf-4 branch with debug build without usb serial

https://user-images.githubusercontent.com/6237028/104805884-a1158900-5816-11eb-90ca-04d68d50f3ef.mp4

**Comment by andycarle** [comment]
Thank you for the report and for trying out our ESP-IDF v4.2 preview!

No, the behavior you're seeing is not expected. It looks like we forgot to pull out a temporary change from the board bring-up process.

I'll get the fix into our next open source push. But in the meantime, it's an easy one to correct locally. 

In [$MODDABLE/xs/platforms/esp/xsPlatform.c](https://github.com/Moddable-OpenSource/moddable/blob/esp-idf-4/xs/platforms/esp/xsPlatform.c#L677) at line 677 you'll see four lines commented out:

```c
//			if (timeout && (timeout < modMilliseconds())) {
//				fxDisconnect(the);
//				break;
//			}
```

Just uncomment those four lines and it should fix the issue.

&nbsp;- Andy

**Comment by phoddie** [comment]
FYI - the ability to run a debug build without the debugger attached is provided as a convenience by the ESP8266 and ESP32 hosts. When no debugger is present, the wait for the debugger introduces a delay in start-up.  As a rule, when running without the debugger you should use release build (`mcconfig -m -p...`) or an instrumented build (`mconfig -i -m -p...`).

**Comment by andycarle** [comment]
This should be fixed by b9119d2. Sorry, I forgot to tag the issue in that commit!

&nbsp; - Andy

**Comment by MKGaru** [comment]
Thank you for fixed. I confirmed.
